src/main/java/reposense/git/GitRemote.java

```diff
@@ -0,0 +1,43 @@
+package reposense.git;
+
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import reposense.system.CommandRunner;
+
+/**
+ * Contains git remote related functionality.
+ * Git remote is responsible for getting remote repository information.
+ */
+public class GitRemote {
+
+    /**
+     * Extracts remote repository information. Returns a map
+     * with keys of the form REMOTE_NAME(fetch) or REMOTE_NAME(push).
```

Comment:
This comment is chronologically after the comment about lines 83 and 85. Are these key names meant to be stored as constants? Because the one in the javadoc does not seem to match with the one used in lines 83 and 85.

Reply:
REMOTE_NAME is meant to be substituted with the name of the remote. Lines 83 and 85 check for the remote 'origin' first if it exists. It could be possible that the repository has other remotes, Am open to suggestions for making the substitution more obvious.

Reply:
oh I understand it now. I think I understand the PR a lot better now. Then I believe there might be a second point of failure for repos where the user renamed the `origin` reference though I guess that is quite a fringe case and the behaviour of reposense would still be acceptable.

Reply:
Yes, if `origin` doesn't exist, we currently get whatever is available.